Support complex schema and discriminator

puneetbehl opened the pull request:
#63 Support complex schema and discriminator

* anyOf, oneOf, and allOf support in responses.
* support for discriminator field and discriminator mapping.

graemerocher commented:
Is this different to https://github.com/micronaut-projects/micronaut-openapi/pull/87 ?

puneetbehl commented:
Yes, this PR supports:

1. The ability to add them under response such as:

```
paths:
  /pets:
    patch:
      requestBody:
        content:
          application/json:
            schema:
              oneOf:
                - $ref: '#/components/schemas/Cat'
                - $ref: '#/components/schemas/Dog'
      responses:
        '200':
          description: Updated
```

I believe earlier we only supported these with `@Schema` annotation i.e. under `components`.

2. In addition, this also adds support for defining `discriminator` property, such as:
```
paths:
  /pets:
    patch:
      requestBody:
        content:
          application/json:
            schema:
              oneOf:
                - $ref: '#/components/schemas/Cat'
                - $ref: '#/components/schemas/Dog'
              discriminator:
                propertyName: pet_type
      responses:
        '200':
          description: Updated
```

fbeli commented:
> Yes, this PR supports:
> 
> 1. The ability to add them under response such as:
> 
> ```
> paths:
>   /pets:
>     patch:
>       requestBody:
>         content:
>           application/json:
>             schema:
>               oneOf:
>                 - $ref: '#/components/schemas/Cat'
>                 - $ref: '#/components/schemas/Dog'
>       responses:
>         '200':
>           description: Updated
> ```
> 
> I believe earlier we only supported these with `@Schema` annotation i.e. under `components`.
> 
> 2. In addition, this also adds support for defining `discriminator` property and discriminator mapping, such as:
> 
> ```
> paths:
>   /pets:
>     patch:
>       requestBody:
>         content:
>           application/json:
>             schema:
>               oneOf:
>                 - $ref: '#/components/schemas/Cat'
>                 - $ref: '#/components/schemas/Dog'
>               discriminator:
>                 propertyName: pet_type
>       responses:
>         '200':
>           description: Updated
> ```

Do you have any examples of how to use it? I cannot get discriminator on my swagger page.
